python_modules/dagster/dagster/core/definitions/inference.py

```diff
@@ -87,11 +169,23 @@ def infer_input_definitions_for_composite_solid(solid_name, fn):
     signature = funcsigs.signature(fn)
     params = list(signature.parameters.values())
 
-    return _infer_inputs_from_params(params, '@composite_solid', solid_name)
+    # try to infer from docstrings
+    defs = _infer_input_definitions_from_docstring(solid_name, fn)
+
+    if not defs:
+        defs = _infer_inputs_from_params(params, '@composite_solid', solid_name)
+
+    return defs
 
 
 def infer_input_definitions_for_solid(solid_name, fn):
     signature = funcsigs.signature(fn)
     params = list(signature.parameters.values())
 
-    return _infer_inputs_from_params(params[1:], '@solid', solid_name)
+    # try to infer from docstrings
+    defs = _infer_input_definitions_from_docstring(solid_name, fn)
+
+    if not defs:
+        defs = _infer_inputs_from_params(params[1:], '@solid', solid_name)
+
+    return defs
```

Comment:
hmm so the tough question here is how to deal with precedence / conflicts. It is not clear to me what to do if someone had python type hints and a rich docstring, especially if those differ. 

curious if @schrockn who filed the original issue has thoughts. We may want to put this docstring parsing behind an explicit flag on the decorators.

Reply:
Hmm, could certainly validate if they match as far as descriptions and dagster types etc but how to notify the user I don't know. Logging a warning probably won't be enough, will prob need something on the UI side so that users can notice easily. Throwing an error would be quick catch....

Reply:
Those are good qs now since there is already explocit Input/Output defs + typings does it even make sense to support docstrings? Like docstrings should prob just describe the purpose of the solids along with the typings defs

Reply:
One path forward would be to only extract *descriptions* from the doc strings and not the typing information. So the set up would be
* if input/output defs explicitly passed in to decorator, use those
* if not infer:
  * type information from python type hints
  * descriptions from docstrings

Reply:
Yeah I like that path , I'll do it that way instead, been busy at work but should have time this weekend (someone else can take this and implement if they want also if I'm taking too long with the codez)